wordpress-mobile/WordPress-Android
# RI: Get user tags only from db if size > 0

This PR 

1. updates`ReaderTagRepository.getUserTags()` to return tags only from the db if count > 0 - d1e5582

    If there are already followed tags in the db, we can skip fetching user tags from the server (and avoid unnecessary full screen error message due to n/w error) as the new discover feed can be populated based on these tags.

2. deletes recommended tags added as followed tags for logged out users inside `GetFollowedTagsUseCase.get()` - fd6c575

    This is done so that they don't cause issues in calculating followed tags count during display of "Choose Interest" screen. 
    More context [here](https://github.com/wordpress-mobile/WordPress-Android/pull/12530#discussion_r462794836).

3.  fixes a loading issue in the discover feed - ef28a4e

#### To Test

Enable RI FF 

1. Launch app and login with account with followed tags
2. Enable airplane mode
3. Go to Reader -> Discover tab 
4. Notice that discover feed is shown and not blocked by any full screen error message

PR submission checklist:
